### PR TITLE
Feature/e2e integration v2.0

### PR DESCRIPTION
**Pull request by samarth18**
Created a new branch (I somehow messed up my local git repo) with all the changes requested in the previous PR. Please review

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -10,15 +10,16 @@ module.exports = {
...
   },
   "rules": {
     "arrow-body-style": "off", // we don't something like (e => e.y) because it makes debugging harder
     "class-methods-use-this": "error",
     "consistent-return": "off",
-    "curly": "error",
+    curly: "error",
     "func-names": "error",
-    "func-style": ["error", "declaration", {"allowArrowFunctions": true}],
+    "func-style": ["error", "declaration", { "allowArrowFunctions": true }],
```
Comment (thezeroalpha) [inline .eslintrc.js]: Make sure the style is consistent, so all keys should be in quotes.

File: graphql-api-gateway-poc/package.json [hunk 2]
```diff
@@ -5,7 +5,6 @@
   "main": "index.js",
   "scripts": {
     "start-gateway": "nodemon src/index.js",
```
Comment (thezeroalpha) [inline graphql-api-gateway-poc/package.json]: This is something I didn't notice earlier, but -- why are you using nodemon here? The Dockerfile is already set up to run pm2 on `src/index.js`. Is there some advantage to using a custom script with nodemon?

Comment (samarth18) [reply graphql-api-gateway-poc/package.json]: This script is not being used for the dockerfile. It was while testing locally. I'll remove it in the next commit. While running the containers we use pm2 only.

**Comment by samarth18** [comment]
> I still don't think the names are very clear. If I see a directory called "products", I would expect that microservice to contain logic for accessing the data of products via REST API requests, when it's actually an Apollo server. The purpose of the code should be clear from the name. So if it's an Apollo server specific to the products microservice, this should be clear from the name, so something like "products-apollo-server" (or "products-apollo"). Don't sacrifice clarity for conciseness.
> 
> Also, I looked a bit into how Apollo works, and it seems to me that it wouldn't be necessary to have an Apollo server for each microservice. It seems to be possible to aggregate data from multiple REST endpoints ([here's what I was reading](https://www.apollographql.com/docs/apollo-server/data/data-sources/)). Could you check that out and add it if possible? Maybe it would be for another ticket/PR.

File: .eslintrc.js (hunk 1 above)
Comment (samarth18) [reply .eslintrc.js]: I made sure that all keys are in quotes. Earlier a plugin in my editor was automatically removing the quotes from keys when the file was saved. I disabled that to make sure quotes are not removed

**Comment by samarth18** [comment]
> I still don't think the names are very clear. If I see a directory called "products", I would expect that microservice to contain logic for accessing the data of products via REST API requests, when it's actually an Apollo server. The purpose of the code should be clear from the name. So if it's an Apollo server specific to the products microservice, this should be clear from the name, so something like "products-apollo-server" (or "products-apollo"). Don't sacrifice clarity for conciseness.
> 
> Also, I looked a bit into how Apollo works, and it seems to me that it wouldn't be necessary to have an Apollo server for each microservice. It seems to be possible to aggregate data from multiple REST endpoints ([here's what I was reading](https://www.apollographql.com/docs/apollo-server/data/data-sources/)). Could you check that out and add it if possible? Maybe it would be for another ticket/PR.

I had already referred to this link earlier and RESTDatasources are already being used by the apollo servers for the microservices to fetch data from the REST API for that microservice. The issue we have is how do we make the gateway communicate directly with the REST API. According to what I have read and the examples I have looked at, the gateway needs a list of graphql services which will communicate with the REST API and this communication is using the RESTDatasources mentioned in the link. I haven't found a similar example where the gateway directly communicates with the REST APIs of the microservices. I am trying some other alternatives to connect the gateway directly with the REST API

File: graphql-api-gateway-poc/package.json (hunk 2 above)
Comment (thezeroalpha) [reply graphql-api-gateway-poc/package.json]: Ah, makes sense.

File: graphql-api-gateway-poc/README.md [hunk 3]
```diff
@@ -2,47 +2,43 @@
...
 
+- The GraphQL API Gateway will be the entrypoint to all requests made by the client to each microservice of the platform
+- Each microservice will have an Apollo server running that will be linked to the RESTful API of that microservice with the help of RESTDataSources from Apollo
+- The Gateway will redirect the request from the client on the basis of the query made to the correct microservice
+- The Apollo server of that microservice will then return the desired result back to the gateway to be returned to the client
+
 ## Installation
 
-To run this demo locally:
+---
```
Comment (DarshitChanpura) [inline graphql-api-gateway-poc/README.md]: remove this

File: members-graphql-service/datasource.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  constructor() {
+    super();
+    // Root domain of the RESTful API for members created by the boilerplate
+    this.baseURL = `http://${host}:${BOILERPLATE_PORT}`;
+  }
+
+  /** 
+  Retrieves all members
+  @returns Array of members
+  */
```
Comment (DarshitChanpura) [inline members-graphql-service/datasource.js]: mention the type of parameters passed and values returned

**Pull request merged**